### PR TITLE
fix: getAccount

### PR DESCRIPTION
**Pull request by greged93**
After checking out the implementation of the endpoint by Geth and Nethermind, I believe we should be returning an `None` value if the account if not found in the state.

## Example: 

```
curl https://docs-demo.quiknode.pro/ \                                                                                                                                                                                                                                    130 ↵
  --header 'Content-Type: application/json' \
  --data '{
      "method": "eth_getAccount",
      "params":["0x8D97689C9818892B700e27F316cc3E41e17fBebf", "latest"],
      "id": 1,
      "jsonrpc": "2.0"
}'
```

Returns `null`.

### CODE REVIEW

File: crates/rpc/rpc-eth-api/src/helpers/state.rs [hunk 1]
```diff
@@ -134,25 +134,26 @@ pub trait EthState: LoadState + SpawnBlocking {
...
 
-            let account = state
-                .basic_account(address)
-                .map_err(Self::Error::from_eth_err)?
-                .unwrap_or_default();
-            let balance = account.balance;
-            let nonce = account.nonce;
-            let code_hash = account.bytecode_hash.unwrap_or(KECCAK_EMPTY);
+            let account = state.basic_account(address).map_err(Self::Error::from_eth_err)?;
+            if let Some(account) = account {
```
Comment (mattsse) [inline crates/rpc/rpc-eth-api/src/helpers/state.rs:142]: styl nit:

let use `let Some() else {return Ok(None)}` here to get rid of the scope

File: crates/rpc/rpc/src/eth/helpers/state.rs [hunk 2]
```diff
@@ -103,10 +100,48 @@ mod tests {
...
+
+    #[tokio::test]
+    async fn test_get_account_empty() {
+        let address = Address::random();
+        let accounts = HashMap::from([(address, ExtendedAccount::new(0, U256::ZERO))]);
+        let eth_api = mock_eth_api(accounts);
+
+        let account = eth_api.get_account(address, Default::default()).await.unwrap();
+        let expected_account =
+            alloy_consensus::Account { code_hash: KECCAK_EMPTY, ..Default::default() };
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/helpers/state.rs:144]: I believe this is available from the reth-rpc-types crate so we don't need the alloy-consensus dep here

File: crates/rpc/rpc/src/eth/helpers/state.rs
Comment (mattsse) [inline crates/rpc/rpc/src/eth/helpers/state.rs:1]: this test case can also be added as an additional call here

https://github.com/paradigmxyz/reth/blob/72ed7a9b0812fb03b87be9e3877713dc06b7dc91/crates/rpc/rpc-builder/tests/it/http.rs#L177-L177

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, smol nits

File: crates/rpc/rpc-eth-api/src/helpers/state.rs [hunk 3]
```diff
@@ -134,25 +134,26 @@ pub trait EthState: LoadState + SpawnBlocking {
...
-            // Provide a default `HashedStorage` value in order to
-            // get the storage root hash of the current state.
-            let storage_root = state
-                .hashed_storage_root(address, Default::default())
-                .map_err(Self::Error::from_eth_err)?;
+                // Provide a default `HashedStorage` value in order to
+                // get the storage root hash of the current state.
+                let storage_root = state
+                    .hashed_storage_root(address, Default::default())
+                    .map_err(Self::Error::from_eth_err)?;
```
Comment (emhane) [inline crates/rpc/rpc-eth-api/src/helpers/state.rs:151]: fix this whitespace pls, try command `make lint`

**Review by emhane — APPROVED** [review]

File: crates/rpc/rpc-eth-api/src/helpers/state.rs (hunk 3 above)
Comment (emhane) [reply crates/rpc/rpc-eth-api/src/helpers/state.rs:151]: nvm, duplicate of https://github.com/paradigmxyz/reth/pull/10402/files#r1723751263

**Review by mattsse — APPROVED** [review]

**Pull request merged**